From 74b87cc4f5035b91dae2e82527dea9804ae1ce4c Mon Sep 17 00:00:00 2001
From: Pedro Palazon Candel <pedro@joyent.com>
Date: Thu, 30 Aug 2018 16:29:53 +0200
Subject: [PATCH] TRITON-591 refreservation not set properly on bhyve disks
 Reviewed by: Jorge Schrauwen <sjorge@blackdot.be> Reviewed by: Mike Gerdts
 <mike.gerdts@joyent.com>

---
 src/vm/node_modules/VM.js | 18 ++----------------
 1 file changed, 2 insertions(+), 16 deletions(-)

diff --git a/src/vm/node_modules/VM.js b/src/vm/node_modules/VM.js
index ac43db55..24b800d4 100644
--- a/src/vm/node_modules/VM.js
+++ b/src/vm/node_modules/VM.js
@@ -2074,23 +2074,7 @@ function createVolume(volume, opts, callback)
 
     if (volume.hasOwnProperty('refreservation')) {
         refreserv = volume.refreservation;
-    } else if (volume.hasOwnProperty('image_uuid')) {
-        // We can't set refreserv > volsize when making a clone, so set it
-        // to volsize instead.
-        refreserv = size;
-        log.debug({
-            size: size
-        }, 'refreservation unset, setting refreserv=size for cloned disk');
-    } else if (brand === 'kvm') {
-        // For KVM we keep the pre-OS-6753 for now to not break compatibility
-        // but could change this to match bhyve (default case below) in the
-        // future.
-        refreserv = size;
-        log.debug({
-            size: size
-        }, 'refreservation unset for kvm, setting refreserv=size for new disk');
     } else {
-        // We default to letting SmartOS set refreservation for bhyve: OS-6753
         log.debug('refreservation unset, letting system determine');
         refreserv = 'auto';
     }
@@ -2146,6 +2130,8 @@ function createVolume(volume, opts, callback)
                 }
                 if (refreserv !== 'auto') {
                     args.push('-o', 'refreservation=' + refreserv + 'M');
+                } else if (refreserv === 'auto') {
+                    args.push('-o', 'refreservation=auto');
                 }
                 args.push(snapshot, target);
                 zfs(args, log, function (e) {
-- 
2.21.0

